PolyEngine/Core/Src/Optional.hpp

```diff
@@ -0,0 +1,188 @@
+#pragma once
+
+#include "Defines.hpp"
+
+namespace Poly {
+	template<typename V>
+	class Optional {
+	public:
+		using ValueType = V;
+	public:
+		Optional() : initialized(false), dummy{} {}
+		Optional(const V&  value) : initialized(true), value_storage(value) {}
+		Optional(      V&& value) : initialized(true), value_storage(std::move(value)) {}
+		~Optional() { if (this->HasValue()) { this->Value().~V(); } }
```

Comment:
Honestly, I don't see a point in using them. They're thin wrappers, that as far as I can see, don't bring anything to the table.